packages/material-ui/src/Button/Button.js

```diff
@@ -227,13 +227,13 @@ function Button(props) {
   } = props;
 
   warning(
-    process.env.MUI_SUPPRESS_DEPRECATION_WARNINGS || variant !== 'flat',
+    getEnv('MUI_SUPPRESS_DEPRECATION_WARNINGS') || variant !== 'flat',
```

Comment:
I think that we can safely remove `MUI_SUPPRESS_DEPRECATION_WARNINGS` here. I have already updated the tests to handle it.